subsys/bluetooth/shell/bt.c

```diff
@@ -1424,9 +1501,74 @@ static int cmd_adv_param(const struct shell *sh, size_t argc, char *argv[])
 	return 0;
 }
 
+static const uint8_t ad_ext_uuid16[] = {
+	IF_ENABLED(CONFIG_BT_ASCS, (BT_UUID_16_ENCODE(BT_UUID_ASCS_VAL),))
+	IF_ENABLED(CONFIG_BT_BASS, (BT_UUID_16_ENCODE(BT_UUID_BASS_VAL),))
+	IF_ENABLED(CONFIG_BT_PACS, (BT_UUID_16_ENCODE(BT_UUID_PACS_VAL),))
+	IF_ENABLED(CONFIG_BT_CAP_ACCEPTOR, (BT_UUID_16_ENCODE(BT_UUID_CAS_VAL),))
+	IF_ENABLED(CONFIG_BT_HAS, (BT_UUID_16_ENCODE(BT_UUID_HAS_VAL),))
+	IF_ENABLED(CONFIG_BT_VCS, (BT_UUID_16_ENCODE(BT_UUID_VCS_VAL),))
+};
+
+static ssize_t ad_ext_init(struct bt_data *data, size_t data_size, bool discoverable)
+{
+	size_t len;
+
+	len = ad_init(data, data_size, discoverable);
+	if (len < 0 || !discoverable) {
+		return len;
+	}
+
+	if (ARRAY_SIZE(ad_ext_uuid16) > 0) {
+		__ASSERT(data_size > len, "No space for AD_UUID16");
+		data[len].type = BT_DATA_UUID16_SOME;
+		data[len].data_len = ARRAY_SIZE(ad_ext_uuid16);
+		data[len].data = &ad_ext_uuid16[0];
+		len++;
+	}
+
+#if defined(CONFIG_BT_AUDIO)
+	if (IS_ENABLED(CONFIG_BT_CAP_ACCEPTOR)) {
+		static const uint8_t ad_cap_announcement[3] = {
+			BT_UUID_16_ENCODE(BT_UUID_CAS_VAL),
+			BT_AUDIO_UNICAST_ANNOUNCEMENT_TARGETED,
+		};
+
+		__ASSERT(data_size > len, "No space for AD_CAP_ANNOUNCEMENT");
+		data[len].type = BT_DATA_SVC_DATA16;
+		data[len].data_len = ARRAY_SIZE(ad_cap_announcement);
+		data[len].data = &ad_cap_announcement[0];
```

Comment:
Since this seems to be a style you've chosen, can you explain why you are doing `&ad_cap_announcement[0]` instead of `ad_cap_announcement`? Both ways work of course, but I'm just curious as to why you prefer the former?

Reply:
I like it better ^^

Reply:
I guess that's a personal observation then :D I have it the other way around, i.e. `data[len].data = ad_cap_announcement` makes me assume that `ad_cap_announcement` is an array (or in any case the data that is being assigned), while `data[len].data = &ad_cap_announcement[0];` indicates, to me, that it's a specific element of an array that is being used. 

But I'll resolve my comments as we don't have a clear guide on this (and I am unaware of any MISRA rules related to this).